course_dashboard/stats.py

```diff
@@ -43,8 +44,16 @@ def dateify(date):
             return date
         return datetime.strptime(date, "%Y-%m-%d %H:%M:%S")
 
-    return [(dateify(result[period_name]), result['enrollment_count']) for result in query]
+    enrollments_per_day = [(dateify(result[period_name]), result['enrollment_count']) for result in query]
+    return add_days_with_no_enrollments(enrollments_per_day)
 
+def add_days_with_no_enrollments(enrollments_per_day):
+    current_day = enrollments_per_day[0][0]
+    for index, enrollment_per_day in enumerate(enrollments_per_day):
+        if enrollment_per_day[0] != current_day:
+            enrollments_per_day.insert(index, (current_day, 0))
```

Comment:
Je ne vois pas bien comment ceci peut fonctionner puisqu'on itère sur un tableau qu'on modifie. Il faut probablement changer d'algo.